src/main/java/ui/components/ButtonThemeSwitch.java

```diff
@@ -0,0 +1,55 @@
+package main.java.ui.components;
+
+import com.formdev.flatlaf.FlatDarkLaf;
+import com.formdev.flatlaf.FlatLaf;
+import com.formdev.flatlaf.FlatLightLaf;
+import com.formdev.flatlaf.extras.FlatAnimatedLafChange;
+import org.kordamp.ikonli.fontawesome5.FontAwesomeSolid;
+import org.kordamp.ikonli.swing.FontIcon;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class ButtonThemeSwitch extends JButton {
+    private final FontIcon darkThemeIcon = FontIcon.of(FontAwesomeSolid.MOON, 13);
+    private final FontIcon lightThemeIcon = FontIcon.of(FontAwesomeSolid.SUN, 13);
+    boolean isDarkTheme = FlatLaf.isLafDark();
```

Comment:
each property should explicitly either be specified public or private. I know convention says its privat but explicitly saying its private makes it easier to understand.
```suggestion
    private boolean isDarkTheme = FlatLaf.isLafDark();
```

Reply:
@Gord310, in this case i have to make it public since im using the variable in a test, but yes i forgot to give it a explicit property, thank you for pointing that out!

Reply:
Oh ok, then I think a getter would be better. So that no one changes this variable outside its component.